tests/lib/ringbuffer/src/main.c

```diff
@@ -443,25 +476,76 @@ void test_reset(void)
 	zassert_true(granted == RINGBUFFER_SIZE - 1, NULL);
 }
 
+#ifdef CONFIG_64BIT
+static uint64_t ringbuf_stored[RINGBUFFER_SIZE];
+#else
+static uint32_t ringbuf_stored[RINGBUFFER_SIZE];
+#endif
+
+/**
+ * @brief verify the array stored by ringbuf
+ *
+ * @ingroup lib_ringbuffer_tests
+ *
+ * @see ring_buf_item_put(), ring_buf_item_get()
+ */
+void test_ringbuffer_array_perf(void)
+{
+	struct ring_buf buf_ii;
+	uint32_t input[3] = {0xaa, 0xbb, 0xcc};
+	uint32_t output[6] = {0xff, 0xff, 0, 0, 0, 0};
+	uint16_t type = 0;
+	uint8_t value = 0, size = 3;
+	int count = 0;
+	void *tp;
+
+	/*Init the ringbuf, and be stored in 64/32 bits array*/
+	ring_buf_init(&buf_ii, RINGBUFFER_SIZE, ringbuf_stored);
+
+	zassert_true(ring_buf_item_put(&buf_ii, 1, 2, input, 3) == 0, NULL);
+
+	/*verify the address stored by ringbuf is contiguous*/
+	for (int i = 0; i < 3; i++) {
+		zassert_equal(input[i], buf_ii.buf.buf32[i+1], NULL);
+	}
+
+	/*Offset to the end of the in-use data in the array*/
+	while (output[count] != 0) {
```

Comment:
Uh... why are we doing this?

You're initializing an output buffer on the stack with some non-zero data at the start, skipping over the non-zero bytes (which you just put there!), and then using the suffix of the buffer as the argument to ring_buf_item_get().  Why?  What's the condition being tested here?  The ring buffer doesn't care what's in the memory before the buffer you passed.

I mean, maybe you're intending to test that it won't write before the buffer you passed?  But if so, you don't actually check that condition; the prefix bytes are ignored.

Reply:
Hi @andyross , Uh... maybe I'm misleading by my thinking. Actually, this operation skipping non-zero data of output buffer maybe is clumsy and confusing. And this test point is covered by the other region of this testcase, so I have removed this code you mentioned. Thanks for your comments again.